ReleaseProcessAutomation/Steps/PipelineSteps/PushNewReleaseBranch.cs

```diff
@@ -0,0 +1,31 @@
+using System.Runtime.InteropServices.ComTypes;
+using ReleaseProcessAutomation.Configuration.Data;
+using ReleaseProcessAutomation.Git;
+
+namespace ReleaseProcessAutomation.Steps.PipelineSteps;
+
+public interface IPushNewReleaseBranch
+{
+  void Execute (string releaseBranchName, string firstAncestorBranchName);
+}
+
+public class PushNewReleaseBranch
+    : IPushNewReleaseBranch
+{
+  private readonly IGitClient _gitClient;
+  private readonly Config _config;
+
+  public PushNewReleaseBranch (IGitClient gitClient, Config config)
+  {
+    _gitClient = gitClient;
+    _config = config;
+  }
+  
+  public void Execute (string releaseBranchName, string firstAncestorBranchName)
+  {
+    var remoteNames = _config.RemoteRepositories.RemoteNames;
+    _gitClient.PushToRepos(remoteNames, releaseBranchName);
+    if(!string.IsNullOrEmpty(firstAncestorBranchName))
+      _gitClient.PushToRepos(remoteNames, firstAncestorBranchName);
```

Comment:
It is kind of strange that we pass in two branch names and do the same thing (push) but the second one is optional. We should remove one of the parameters and let the caller decide how often they want to call us. The `ReleaseOnMasterStep` would call us twice while the `ReleasePatchStep` just once.

Reply:
ReleasePatchStep would also need to call this twice, I think the only thing we could/should do is to remove the optionality of the second argument.

Reply:
If the second parameter is no longer optional then there is no reason why the method has to push two branches at the same time. In either case I think it is better to have the functionality of "pushing a branch" rather then "pushing a branch and optionally another one as well". I think this is better because the user can decide what they want to do. Otherwise, if you just want to push a single branch you either have to duplicate the logic or pass an empty string just for the sake of providing values.